integration_test/src/java/com/twitter/heron/integration_test/core/IntegrationTestSpout.java

```diff
@@ -134,12 +134,11 @@ public void nextTuple() {
 
   @Override
   public void ack(Object messageId) {
-    LOG.fine("Received a ack with MessageId: " + messageId);
-
-    tuplesToAck--;
     if (!isTestMessageId(messageId)) {
       delegateSpout.ack(messageId);
     } else {
+      tuplesToAck--;
```

Comment:
tuplesToAck gets incremented when a test message is sent, but decremented when any ack is recieved, even if it's not a test message. This seems like a bug, so I changed decrement to only happen when acks for test messages are received.

I actually don't know why we need to track tuplesToAck, and why we can't just check that size of the pendingMessages map gets to 0, but I'm not ready to make that change yet.